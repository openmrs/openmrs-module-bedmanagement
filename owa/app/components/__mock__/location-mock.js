Object.defineProperties(window.location, {
    host : {
        writable: true,
        value: '192.168.33.10'
    },
    hostname : {
        writable: true,
        value: '192.168.33.10'
    },
    href :{
        writable: true,
        value: 'https://192.168.33.10/openmrs/owa/bedmanagement/admissionLocations.html'
    },
    origin :{
        writable: true,
        value: 'https://192.168.33.10'
    },
    pathname :{
        writable: true,
        value: '/openmrs/owa/bedmanagement/admissionLocations.html'
    },
    port :{
        writable: true,
        value: ''
    },
    protocol :{
        writable: true,
        value: 'https:'
    }
});