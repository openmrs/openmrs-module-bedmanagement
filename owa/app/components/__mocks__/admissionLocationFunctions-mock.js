import pickBy from 'lodash.pickby';
import {admissionLocations, visitLocations, bedTypes} from 'utilities/__tests__/testData.json';

const admissionLocationFunctionsMock = {
    setActiveLocationUuid: (admissionLocationUuid) => jest.fn(),
    getActiveLocationUuid: () => {
        return 'baf7bd38-d225-11e4-9c67-080027b662ec';
    },
    setState: (data) => jest.fn(),
    getState: () => jest.fn(),
    getAdmissionLocations: () => {
        return admissionLocations;
    },
    getVisitLocations: () => {
        return visitLocations;
    },
    getBedTypes: () => {
        return bedTypes;
    },
    getAdmissionLocationByUuid: (admissionLocationUuid) => {
        return typeof admissionLocations[admissionLocationUuid] !== 'undefined' ?
            admissionLocations[admissionLocationUuid] : null;
    },
    getParentAdmissionLocation: (admissionLocationUuid) => {
        const parentAdmissionLocationUuid = admissionLocations[admissionLocationUuid].parentAdmissionLocationUuid;
        return parentAdmissionLocationUuid != null ? admissionLocations[parentAdmissionLocationUuid] : null;
    },
    getChildAdmissionLocations : (admissionLocationUuid) => {
        return  pickBy(admissionLocations, (admissionLocation) => {
            return admissionLocation.parentAdmissionLocationUuid == admissionLocationUuid;
        });
    },
    reFetchAllAdmissionLocations : () => jest.fn(),
    notify : (notifyType, message) => jest.fn()
};

export default admissionLocationFunctionsMock;
