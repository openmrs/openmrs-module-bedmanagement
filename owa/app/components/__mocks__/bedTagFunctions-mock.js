import _ from 'lodash';

import {bedTags} from 'utilities/__tests__/testData.json';

const bedTagFunctionsMock = {
    setState: (stateData) =>  jest.fn(),
    getBedTags: () => {
        return bedTags;
    },
    getBedTagByUuid: (bedTagUuid) => {
        return _.find(bedTags, function (bedTag) {
            return bedTag.uuid == bedTagUuid;
        });
    },
    fetchBedTags: () => jest.fn(),
    notify : (notifyType, message) => jest.fn()
};

export default bedTagFunctionsMock;
