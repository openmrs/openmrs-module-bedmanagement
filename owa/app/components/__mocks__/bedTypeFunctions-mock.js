import _ from 'lodash';

import {bedTypes} from 'utilities/__tests__/testData.json';

const bedTypeFunctionsMock = {
    setState: (stateData) =>  jest.fn(),
    getBedTypes: () => {
        return bedTypes;
    },
    getBedTypeById: (bedTypeId) => {
        return _.find(bedTypes, function (bedType) {
            return bedType.id == bedTypeId;
        });
    },
    getBedTypeName: (bedTypeName) => {
        return _.find(bedTypes, function (bedType) {
            return bedType.name == bedTypeName;
        });
    },
    fetchBedTypes: () => {
        return bedTypes;
    },
    notify : (notifyType, message) => jest.fn()
};

export default bedTypeFunctionsMock;
