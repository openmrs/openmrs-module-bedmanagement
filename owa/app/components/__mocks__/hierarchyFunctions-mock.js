import {admissionLocations} from 'utilities/__tests__/testData.json';

const hierarchyFunctionMock = {
    getActiveUuid: () => {
        return 'baf7bd38-d225-11e4-9c67-080027b662ec';
    },
    getAdmissionLocations: () => {
        return admissionLocations;
    },
    setActiveUuid: (locationUuid) => jest.fn(),
    setState: (stateData) => jest.fn(),
    toggleIsOpen: (locationUuid, isOpen) => jest.fn(),
    setAdmissionLocationIsOpen: (locationUuid, isOpen) => jest.fn()
};

export default hierarchyFunctionMock;