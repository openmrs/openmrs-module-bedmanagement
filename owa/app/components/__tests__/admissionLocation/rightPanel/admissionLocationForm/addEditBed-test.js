import React from 'react';
import {shallow, mount} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

import AddEditBed from 'components/admissionLocation/rightPanel/admissionLocationForm/addEditBed';
import admissionLocationFunctionsMock from 'components/__mocks__/admissionLocationFunctions-mock';

require('components/__mocks__/location-mock');
const testProps = {
    admissionLocationFunctions: admissionLocationFunctionsMock,
    admissionLocations: admissionLocationFunctionsMock.getAdmissionLocations(),
    addBedData: {
        bedId: null,
        bedNumber: null,
        bedType:null,
        bedUuid: null,
        columnNumber: 2,
        rowNumber: 1,
        status: null,
        layoutColumn: 4,
        layoutRow: 3
    },
    editBedData: {
        bedId: 42,
        bedNumber: '100-a',
        bedType:{
            description: 'LXY',
            displayName: 'luxury bed',
            id: 2,
            name: 'luxury'
        },
        bedUuid: 'a3e42812-a4c1-4453-96cf-d45578d0fea9',
        columnNumber: 1,
        rowNumber: 2,
        status: 'AVAILABLE',
        layoutColumn: 3,
        layoutRow: 4
    }
};

describe('AddEditBed', () => {
    beforeAll(() => {
        var mock = new MockAdapter(axios);
        const data = {
            'id': 53,
            'uuid': 'a3e42812-a4c1-4453-96cf-d45578d0fea9',
            'bedNumber': '100-A',
            'bedType': {
                'id': 1,
                'name': 'deluxe',
                'displayName': 'deluxe bed',
                'description': 'DLX'
            },
            'row': 2,
            'column': 1,
            'status': 'AVAILABLE'
        };

        mock.onPost('https://192.168.33.10/openmrs/ws/rest/v1/bed/a3e42812-a4c1-4453-96cf-d45578d0fea9')
            .reply(200, data);
    });

    it('Should display add edit bed form properly', () => {
        const addBedForm = shallow(<AddEditBed operation='add' layoutColumn={4} layoutRow={3}
            activeUuid='baf7bd38-d225-11e4-9c67-080027b662ec' bed={testProps.addBedData}
            bedTypes={testProps.admissionLocationFunctions.getBedTypes()}
            admissionLocationFunctions={testProps.admissionLocationFunctions}/>);

        expect(addBedForm.find('#location-name').text().trim()).toBe('General Ward');
        expect(addBedForm.find('#row-field').props().value).toBe(1);
        expect(addBedForm.find('#column-field').props().value).toBe(2);
        expect(addBedForm.find('#bed-number-field').props().value).toBe('');
        expect(shallowToJson(addBedForm)).toMatchSnapshot();


        const editBedForm = shallow(<AddEditBed operation='edit' layoutColumn={4} layoutRow={3}
            activeUuid='baf7bd38-d225-11e4-9c67-080027b662ec' bed={testProps.editBedData}
            bedTypes={testProps.admissionLocationFunctions.getBedTypes()}
            admissionLocationFunctions={testProps.admissionLocationFunctions}/>);

        expect(editBedForm.find('#location-name').text().trim()).toBe('General Ward');
        expect(editBedForm.find('#row-field').props().value).toBe(2);
        expect(editBedForm.find('#column-field').props().value).toBe(1);
        expect(editBedForm.find('#bed-number-field').props().value).toBe('100-a');
        expect(editBedForm.find('#bed-type').props().value).toBe('luxury');
        expect(shallowToJson(editBedForm)).toMatchSnapshot();
    });

    it('Should trigger event handler', () => {
        const spyOnCancelEventHandler = jest.spyOn(AddEditBed.prototype, 'cancelEventHandler');
        const spyOnSubmitHandler = jest.spyOn(AddEditBed.prototype, 'onSubmitHandler');
        const spyOnChangeRowField = jest.spyOn(AddEditBed.prototype, 'onChangeRowField');
        const spyOnChangeColumnField = jest.spyOn(AddEditBed.prototype, 'onChangeColumnField');
        const spnOnChangeBedNumberField = jest.spyOn(AddEditBed.prototype, 'onChangeBedNumberField');
        const spyOnChangeBedType = jest.spyOn(AddEditBed.prototype, 'onChangeBedType');
        const spySetState = jest.spyOn(testProps.admissionLocationFunctions, 'setState');
        const editBedForm = mount(<AddEditBed operation='edit' layoutColumn={4} layoutRow={3}
            activeUuid='baf7bd38-d225-11e4-9c67-080027b662ec' bed={testProps.editBedData}
            bedTypes={testProps.admissionLocationFunctions.getBedTypes()}
            admissionLocationFunctions={testProps.admissionLocationFunctions}/>);

        editBedForm.find('#row-field').simulate('change');
        expect(spyOnChangeRowField).toHaveBeenCalled();
        editBedForm.find('#column-field').simulate('change');
        expect(spyOnChangeColumnField).toHaveBeenCalled();
        editBedForm.find('#bed-number-field').simulate('change');
        expect(spnOnChangeBedNumberField).toHaveBeenCalled();
        editBedForm.find('#bed-type').simulate('change');
        expect(spyOnChangeBedType).toHaveBeenCalled();

        editBedForm.find('.btn-danger').simulate('click');
        expect(spyOnCancelEventHandler).toHaveBeenCalled();
        expect(spySetState).toHaveBeenCalledWith({
            activePage: 'listing',
            pageData: {},
            activeUuid: 'baf7bd38-d225-11e4-9c67-080027b662ec'
        });

        editBedForm.find('input[type=\'submit\']').simulate('submit');
        expect(spyOnSubmitHandler).toHaveBeenCalled();

    });
});