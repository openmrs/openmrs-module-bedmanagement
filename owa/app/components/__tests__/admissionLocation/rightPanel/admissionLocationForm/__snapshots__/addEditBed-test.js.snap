// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddEditBed Should display add edit bed form properly 1`] = `
<div
  className="main-container"
>
  <div
    className="admission-location-form"
  >
    <div
      className="block"
    >
      <div
        className="block-title"
      >
        Add
         Bed
        <a
          className="back-link"
          href="javascript:void(0);"
          onClick={[Function]}
          title="back"
        >
          &lt;&lt; Back
        </a>
      </div>
      <div
        className="block-content"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-block"
          >
            <label
              className="form-title"
            >
              Location
            </label>
            <span
              id="location-name"
            >
              General Ward
            </span>
          </div>
          <div
            className="form-block"
          >
            <label
              className="form-title"
            >
              Row
            </label>
            <input
              id="row-field"
              onChange={[Function]}
              required={true}
              type="number"
              value={1}
            />
          </div>
          <div
            className="form-block"
          >
            <label
              className="form-title"
            >
              Column
            </label>
            <input
              id="column-field"
              onChange={[Function]}
              required={true}
              type="number"
              value={2}
            />
          </div>
          <div
            className="form-block"
          >
            <label
              className="form-title"
            >
              Bed Number
            </label>
            <input
              id="bed-number-field"
              onChange={[Function]}
              required={true}
              type="text"
              value=""
            />
          </div>
          <div
            className="form-block"
          >
            <label
              className="form-title"
            >
              Bed Type
            </label>
            <select
              id="bed-type"
              onChange={[Function]}
              required={true}
              value="deluxe"
            >
              <option
                key="1"
                value="deluxe"
              >
                deluxe
              </option>
              <option
                key="2"
                value="luxury"
              >
                luxury
              </option>
              <option
                key="3"
                value="dbl"
              >
                dbl
              </option>
            </select>
          </div>
          <div
            className="form-block"
          >
            <input
              className="btn btn-primary float-left margin-right"
              disabled={false}
              name="submit"
              type="submit"
              value="Save"
            />
            <input
              className="btn btn-danger float-left"
              name="submit"
              onClick={[Function]}
              type="button"
              value="Cancel"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`AddEditBed Should display add edit bed form properly 2`] = `
<div
  className="main-container"
>
  <div
    className="admission-location-form"
  >
    <div
      className="block"
    >
      <div
        className="block-title"
      >
        Edit
         Bed
        <a
          className="back-link"
          href="javascript:void(0);"
          onClick={[Function]}
          title="back"
        >
          &lt;&lt; Back
        </a>
      </div>
      <div
        className="block-content"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-block"
          >
            <label
              className="form-title"
            >
              Location
            </label>
            <span
              id="location-name"
            >
              General Ward
            </span>
          </div>
          <div
            className="form-block"
          >
            <label
              className="form-title"
            >
              Row
            </label>
            <input
              id="row-field"
              onChange={[Function]}
              required={true}
              type="number"
              value={2}
            />
          </div>
          <div
            className="form-block"
          >
            <label
              className="form-title"
            >
              Column
            </label>
            <input
              id="column-field"
              onChange={[Function]}
              required={true}
              type="number"
              value={1}
            />
          </div>
          <div
            className="form-block"
          >
            <label
              className="form-title"
            >
              Bed Number
            </label>
            <input
              id="bed-number-field"
              onChange={[Function]}
              required={true}
              type="text"
              value="100-a"
            />
          </div>
          <div
            className="form-block"
          >
            <label
              className="form-title"
            >
              Bed Type
            </label>
            <select
              id="bed-type"
              onChange={[Function]}
              required={true}
              value="luxury"
            >
              <option
                key="1"
                value="deluxe"
              >
                deluxe
              </option>
              <option
                key="2"
                value="luxury"
              >
                luxury
              </option>
              <option
                key="3"
                value="dbl"
              >
                dbl
              </option>
            </select>
          </div>
          <div
            className="form-block"
          >
            <input
              className="btn btn-primary float-left margin-right"
              disabled={false}
              name="submit"
              type="submit"
              value="Save"
            />
            <input
              className="btn btn-danger float-left"
              name="submit"
              onClick={[Function]}
              type="button"
              value="Cancel"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
